tests/epyccel/test_arrays.py

```diff
@@ -3474,6 +3474,85 @@ def test_iterate_slice(language):
     i = randint(2, 10)
     assert f1(i) == f2(i)
 
+#==============================================================================
+# TESTS: VIEW
+#==============================================================================
+
+@pytest.mark.parametrize( 'language', [
+        pytest.param("fortran", marks = [
+            pytest.mark.skip(reason="strides not implemented in fortran"),
+            pytest.mark.fortran
+        ]),
+        pytest.param("c", marks = pytest.mark.c)
+    ]
+)
+def test_array_1d_view_assign(language):
+    f1 = arrays.array_1d_view_assign
+    f2 = epyccel(f1, language = language)
+    x1 = np.ones((10,), dtype=int)
+    x2 = np.ones((10,), dtype=int)
+    f1(x1[:], 2)
+    f2(x2[:], 2)
+    assert np.array_equal(x1, x2)
+    f1(x1[:3:], 3)
+    f2(x2[:3:], 3)
+    assert np.array_equal(x1, x2)
+    f1(x1[6::], 4)
+    f2(x2[6::], 4)
+    assert np.array_equal(x1, x2)
+
+@pytest.mark.parametrize( 'language', [
+        pytest.param("fortran", marks = [
+            pytest.mark.skip(reason="strides not implemented in fortran"),
+            pytest.mark.fortran
+        ]),
+        pytest.param("c", marks = pytest.mark.c)
+    ]
+)
+def test_array_2d_view_assign(language):
+    f1 = arrays.array_1d_view_assign
+    f2 = epyccel(f1, language = language)
+    x1 = np.ones((5, 5), dtype=int)
+    x2 = np.ones((5, 5), dtype=int)
+    f1(x1[0, :], 2)
+    f2(x2[0, :], 2)
+    assert np.array_equal(x1, x2)
+    f1(x1[:, 3], 2)
+    f2(x2[:, 3], 2)
+    assert np.array_equal(x1, x2)
+
+@pytest.mark.parametrize( 'language', [
+        pytest.param("fortran", marks = [
+            pytest.mark.skip(reason="strides not implemented in fortran"),
+            pytest.mark.fortran
+        ]),
+        pytest.param("c", marks = pytest.mark.c)
+    ]
+)
+def test_array_3d_view_assign(language):
+    f1 = arrays.array_1d_view_assign
+    f2 = epyccel(f1, language = language)
+    x1 = np.ones((3, 4, 5), dtype=int)
+    x2 = np.ones((3, 4, 5), dtype=int)
+    f1(x1[0, 1, :], 2)
+    f2(x2[0, 1, :], 2)
+    assert np.array_equal(x1, x2)
+    f1(x1[2, :, 3], 3)
+    f2(x2[2, :, 3], 3)
+    assert np.array_equal(x1, x2)
+    f1(x1[:, 2, 4], 4)
+    f2(x2[:, 2, 4], 4)
+    assert np.array_equal(x1, x2)
+    f1(x1[::2, 3, 4], 5)
+    f2(x2[::2, 3, 4], 5)
+    assert np.array_equal(x1, x2)
+    f1(x1[2, ::3, 4], 6)
+    f2(x2[2, ::3, 4], 6)
```

Comment:
Could you please add a test with a slice of length 1

Reply:
https://github.com/pyccel/pyccel/pull/1299/commits/8a6e2b3cd1ae93ebef9c0a56fcc55f1efcee9c5c